nav-app/src/app/tabs/tabs.component.ts

```diff
@@ -545,22 +545,16 @@ export class TabsComponent implements AfterContentInit {
             var string = String(reader.result);
             try{
                 viewModel.deSerializeDomainID(string);
-
+                if (!this.dataService.getDomain(viewModel.domainID)) {
+                    throw {message: "Error: '" + viewModel.domain + "' (" + viewModel.version + ") is an invalid domain."};
+                }
                 this.versionUpgradeDialog(viewModel).then( () => {
-                    if (!this.dataService.getDomain(viewModel.domainID)) {
-                        throw {message: "Error: '" + viewModel.domain + "' (" + viewModel.version + ") is an invalid domain."};
-                    }
+                    this.openTab("new layer", this.layerTab, viewModel, true, true, true, true);
```

Comment:
👍  I like that it opens the tab first, then loads the data. Better than what I was originally thinking (putting the spinner over the new tab interface for the duration of the load).